### DISCUSSION TITLE
Highlight method exit points instead of type name occurrences?

### DISCUSSION THREAD

**Discussion by ctjdr** [redhat-developer/vscode-java#2193] [Ideas]
Some IDEs like Eclipse provide a feature I find pretty handy and which I would love to see in VS Code:
If I click on a method's return type declaration, the IDE highlights any exit point in the according method body. This includes all `return` statements, `throw` statements, and method calls that could throw an unchecked exception.
In addition clicking on the `return` keyword highlights all other exit points inside the same method.

Currently, if I click on the return type declaration in VS Code it highlights every occurrence of that type name within the same file, which I don't find that useful.

Are there any plans to implement this? Or is the current behavior intended by design and should not change?

Thanks!

P.S.: You do a great job improving Java support in VS Code. Really appreciate it!

**Comment by 0dinD** [comment]
Yeah this would be really handy actually. Since the extension uses a language server based on Eclipse libraries, many features in the Eclipse IDE are fairly straightforward to port. I looked into this one, and it seems pretty simple, so I might submit a PR to implement this when I have time.

**Reply by 0dinD** [reply]
Here, I just implemented it:

![image](https://user-images.githubusercontent.com/52179873/139740594-1c813e4d-6ff4-4078-874a-3bff2c65d60c.png)

Is that what you wanted? I'll need to clean up the code first, so we'll see when I have time to submit a PR.

**Reply by ctjdr** [reply]
@0dinD, this is great! Exactly what I was after. Looking forward to see the PR getting into the project.

**Reply by 0dinD** [reply]
@ctjdr I've submitted eclipse/eclipse.jdt.ls#1941 now, which you can follow to get notified when it gets merged.

**Reply by ctjdr** [reply]
@0dinD, today while coding I noticed the feature has already arrived in VS Code (most probably with the Feburary update, v1.65). I very much appreciate the effort you put into it. This is fantastic. Thanks!
